src/Core/Workspace/Project.cs

```diff
@@ -24,19 +25,26 @@ internal static class ProjectExtensions
 
     internal class ProjectFileComparer : EqualityComparer<Project>
     {
-        public override bool Equals(Project p1, Project p2) => p1.ProjectFile == p2.ProjectFile;
+        private readonly StringComparison StringComparison =
+            RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX)
```

Comment:
Is Windows/OSX handling project file names differently?

Reply:
By default, file systems are case-insensitive in Windows and OS X, and case-sensitive on Linux. So this is just used as a best-effort to determine if two paths are the "same" or not. It's not going to be 100% accurate, but I think it's the best we can do without a lot of extra effort. (You would think there is some built-in .NET functionality to do this, but there was nothing I could find...)